src/apps/chifra/internal/scrape/handle_scrape.go

```diff
@@ -115,12 +122,10 @@ func (opts *ScrapeOptions) HandleScrape() error {
 			if !ok {
 				break
 			}
-			goto PAUSE
+			continue
 		}
-
-	PAUSE:
-		opts.Pause(progress)
 	}
 
-	return nil
+	// should never append (see caller)
+	return errors.New("unexpected loop exit")
```

Comment:
There are many cases where the loop will return an error (for example, the node software died). We shouldn't leave the loop simply because of an error. Perhaps there's a better way to distinguish why the error occurred, but simply leaving this loop isn't the right solution, I don't think.